File: API/AmusementPark.Core/Domain/Parks/StandaloneAttraction.cs
using AmusementPark.Core.Geo;
using AmusementPark.Core.Localization;

namespace AmusementPark.Core.Domain.Parks;

/// <summary>
/// Attraction fixe publiable sans parc parent.
/// </summary>
public sealed class StandaloneAttraction : GeolocatedEntityBase
{
    public string Name { get; set; } = string.Empty;

    public string? CountryCode { get; set; }

    public ParkItemType Type { get; set; } = ParkItemType.Attraction;

    public string? Subtype { get; set; }

    public string? OperatorId { get; set; }

    public string? WebsiteUrl { get; set; }

    public string? Street { get; set; }

    public string? City { get; set; }

    public string? PostalCode { get; set; }

    public List<LocalizedText> Descriptions { get; set; } = new();

    public AttractionDetails? AttractionDetails { get; set; }

    public AttractionLocations? AttractionLocations { get; set; }

    public bool IsVisible { get; set; }

    public AdminReviewStatus AdminReviewStatus { get; set; } = AdminReviewStatus.ToReview;

    public string? LegacyParkId { get; set; }

    public string? LegacyParkItemId { get; set; }

    public bool IsPubliclyPublishable()
    {
        return !string.IsNullOrWhiteSpace(this.Id)
            && !string.IsNullOrWhiteSpace(this.Name)
            && this.IsVisible
            && this.AdminReviewStatus != AdminReviewStatus.NotRelevant
            && !ParkItemStatusNormalizer.IsClosedDefinitively(this.AttractionDetails?.Status);
    }
}

File: FRONT/AmusementPark/src/app/models/standalone-attractions/standalone-attraction.ts
import { AdminReviewStatus } from '@app/models/admin/admin-review-status';
import { AttractionDetails } from '@app/models/parks/attraction-details';
import { AttractionLocations } from '@app/models/parks/attraction-locations';
import { ParkItemType } from '@app/models/parks/park-item-type';
import { LocalizedItem } from '@app/models/shared/localized-item';

export interface StandaloneAttraction {
  id?: string | null;
  name: string;
  countryCode?: string | null;
  type: ParkItemType;
  subtype?: string | null;
  operatorId?: string | null;
  websiteUrl?: string | null;
  street?: string | null;
  city?: string | null;
  postalCode?: string | null;
  latitude?: number | null;
  longitude?: number | null;
  descriptions?: LocalizedItem<string>[];
  attractionDetails?: AttractionDetails | null;
  attractionLocations?: AttractionLocations | null;
  isVisible: boolean;
  adminReviewStatus: AdminReviewStatus;
  legacyParkId?: string | null;
  legacyParkItemId?: string | null;
}

export interface StandaloneAttractionMigrationRequest {
  legacyParkId: string;
  legacyParkItemId?: string | null;
  targetStandaloneAttractionId?: string | null;
  retireLegacyPark: boolean;
  retireLegacyParkItem: boolean;
}
